statsmodels/distributions/discrete.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+from scipy.stats import rv_discrete
+from scipy.special import gammaln
+
+class genpoisson_p_gen(rv_discrete):
+    '''A generalized Poisson discrete random variable
+    '''
+    def  _rvs(self, mu, alpha, p):
+        pass
+
+    def _logpmf(self, Y, mu, alpha, p):
+        mu_p = mu ** (p - 1)
+        a1 = 1 + alpha * mu_p
+        a2 = mu + (a1 - 1) * Y
+        return (np.log(mu) + (Y - 1) * np.log(a2) - Y *
+                np.log(a1) - gammaln(Y + 1) - a2 / a1)
```

Comment:
cosmetic: use a temp variable instead of a multi-line return and make line breaks at + or - not higher precedence terms like * or /

Also AFAIR, inline compositions should work
pmf_ = np.log(mu) + (Y - 1) * np.log(a2)
pmf_ += y * np.log(a1) - gammaln(Y + 1) - a2 / a1

no capital letters and instead of Y the scipy distribution pattern might be `x`